ipfsconn/ipfshttp/ipfshttp.go

```diff
@@ -450,6 +483,79 @@ func checkResponse(path string, code int, body []byte) error {
 	return fmt.Errorf("IPFS-post '%s' unsuccessful: %d: %s", path, code, body)
 }
 
+func handleRefsProgress(dec *json.Decoder, reset chan int) error {
+	var last string
+	for {
+		var ref ipfsRefsResp
+		if err := dec.Decode(&ref); err == io.EOF {
+			break
+		} else if err != nil {
+			return err
+		}
+		//logger.Infof("Ref %s", ref.Ref)
+		if ref.Err != "" {
+			logger.Error(ref.Err)
+		}
+
+		if ref.Ref != last {
+			last = ref.Ref
+			reset <- 1
+		}
+	}
+
+	return nil
+}
+
+func handlePinsProgress(dec *json.Decoder, reset chan int) error {
+	var progress int
+	for {
+		var pins ipfsPinsResp
+		if err := dec.Decode(&pins); err == io.EOF {
+			break
+		} else if err != nil {
+			return err
+		}
+		//logger.Infof("Progress: %d", pins.Progress)
+		if pins.Progress > progress {
+			progress = pins.Progress
+			reset <- 1
+		}
+	}
+
+	return nil
+}
+
+func checkTimeout(ctx context.Context, cancel context.CancelFunc, timer *time.Timer, timeout time.Duration, reset chan int) {
```

Comment:
All parameters are used

Reply:
ah ok, but the timer could just be created inside passing only the timeout ?

Reply:
you can probably avoid passing the context by closing the reset channel and checking if it has been closed with `case _, ok := reset:`

Reply:
Using context let's me take care of the situation when we don't timeout and context is Done, because of something. May be this can be done without passing context, but why not pass context?